xds/internal/balancer/ringhash/ring.go

```diff
@@ -122,24 +122,21 @@ func newRing(subConns *resolver.AddressMap, minRingSize, maxRingSize uint64, log
 // Must be called with a non-empty subConns map.
 func normalizeWeights(subConns *resolver.AddressMap) ([]subConnWithWeight, float64) {
 	var weightSum uint32
-	keys := subConns.Keys()
-	for _, a := range keys {
-		weightSum += getWeightAttribute(a)
+	scVals := subConns.Values()
+	for _, a := range scVals {
+		weightSum += a.(*subConn).weight
 	}
```

Comment:
Maybe a one line comment above this block saying we are using `Keys` because attributes (which contains address weights) are removed from `Values`? 

This took a bit of code reading and direct conversation with you to figure out what is happening. So, a one liner would help future readers of the code.

Reply:
Make sense. I added.